shared/util/saga.js

```diff
@@ -23,20 +45,23 @@ function* sequentially(effects: Array<any>): Generator<any, Array<any>, any> {
 // TODO I couldn't get flow to figure out how to infer this, or even force you to explicitly do it
 // maybe flow-strict fixes this
 type MaybeAction = void | boolean | TypedActions | null
-function* chainAction<Actions>(
+function* chainAction<Actions: {+type: string}>(
   pattern: RS.Pattern,
   f: (
     state: TypedState,
-    action: Actions
-  ) => MaybeAction | $ReadOnlyArray<MaybeAction> | Promise<MaybeAction | $ReadOnlyArray<MaybeAction>>
+    action: Actions,
+    logger: SagaLogger
+  ) => MaybeAction | $ReadOnlyArray<MaybeAction> | Promise<MaybeAction | $ReadOnlyArray<MaybeAction>>,
+  fcnTag?: string // tag for logger
 ): Generator<any, void, any> {
   type Fn = Actions => RS.Saga<void>
   return yield Effects.takeEvery<Actions, void, Fn>(pattern, function* chainActionHelper(
     action: Actions
   ): RS.Saga<void> {
+    const sl = new SagaLogger(action.type, fcnTag || 'unknown')
```

Comment:
At first I thought maybe we could use f.name, but then I realized that would be useless when we minified it.

Reply:
Yeah it was originally that :